### PR TITLE
polyglot: init at version 3.5.1

### LINKED ISSUE #355496: Package request: PolyGlot
**Project description**

PolyGlot is a tool that is designed to help in the design, creation, and publication of constructed languages, or "conlangs."

**Metadata**

* homepage URL: http://draquet.github.io/PolyGlot/index.html
* source URL: https://github.com/DraqueT/PolyGlot
* license: MIT
* platforms: Windows, Mac, Linux (deb)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Noodlez1232**
## Description of changes

Added a package for PolyGlot, a conlang toolkit.
Homepage is here: https://draquet.github.io/PolyGlot/readme.html

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/po/polyglot/package.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  makeWrapper,
+  maven,
+  libGL,
+  xdg-utils,
+  zip,
+  zlib
+}:
+maven.buildMavenPackage rec {
+  pname = "polyglot";
+  version = "3.5.1";
```
Comment (jtbx) [inline pkgs/by-name/po/polyglot/package.nix:14]:
```suggestion
  version = "3.5.1";

```

File: pkgs/by-name/po/polyglot/package.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      --add-flags "-Djpackage.app-version=${version}" \
+      --add-flags "-jar $out/share/PolyGlotLinA/PolyGlotLinA-${version}-jar-with-dependencies.jar"
+  '';
+
+  meta = {
+    description = "A conlang construction toolkit";
+    homepage = "https://draquet.github.io/PolyGlot/readme.html";
+    changelog = "https://github.com/DraqueT/PolyGlot/releases/tag/${version}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.noodlez1232 ];
```
Comment (jtbx) [inline pkgs/by-name/po/polyglot/package.nix:50]: Makes it easier for other maintainers to join if they want
```suggestion
    maintainers = with lib.maintainers; [ noodlez1232 ];
```

File: pkgs/by-name/po/polyglot/package.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    platforms = lib.platforms.linux;
```
Comment (jtbx) [inline pkgs/by-name/po/polyglot/package.nix:51]: The PolyGlot website has binaries for macOS too
```suggestion
    platforms = lib.platforms.all;
```

**Review by jtbx — CHANGES_REQUESTED** [review]

File: pkgs/by-name/po/polyglot/package.nix (hunk 3 above)
Comment (Noodlez1232) [reply pkgs/by-name/po/polyglot/package.nix:51]: I don't have a Mac, and the build process for Mac is different, so I'm not sure if I could figure out how to build it. If you have any advice on that, I'd appreciate it.

Comment (Noodlez1232) [reply pkgs/by-name/po/polyglot/package.nix]: I decided to just push it with it set to `.all` and we'll see. I'll debug using CI.

**Review by jtbx — APPROVED** [review]
Yeah this looks good. If you can't get it to build on darwin just set meta.platforms back to linux.

@ofborg build polyglot

**Comment by Noodlez1232** [comment]
Is there any way we can get this pushed? It seems good to me.

**Comment by Noodlez1232** [comment]
Actually I take that back. It seems there's different maven deps for darwin, so I'll just remove that. If I get a darwin box to test these things on, I'll revisit it.

**Comment by Noodlez1232** [comment]
This is ready to merge. I will work on the darwin capability later. I have a VM working, but I also need to check aarch64-linux and aarch64-darwin as well.

**Review by jtbx — CHANGES_REQUESTED** [review]
CI is failing because your code is incorrectly formatted. If you run `nix-shell` in the root of your nixpkgs tree it should tell you how to fix it.

**Comment by Noodlez1232** [comment]
@jtbx There we go just fixed it... I thought I had formatted it before my bad.

**Review by jtbx — APPROVED** [review]
Thank you!

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
There is an error on startup
```
java.lang.UnsatisfiedLinkError: /home/user/.openjfx/cache/14/libprism_es2.so: libGL.so.1: cannot open shared object file: No such file or directory
```
But it seems to not affect the program much.

**Comment by Noodlez1232** [comment]
> There is an error on startup
> [...]

Yeah I've been trying to figure it out since I packaged it. I just learned about the lib.makeLibraryPath function and all that, so I actually was able to fix it by explicitly adding it to the LD_LIBRARY_PATH.

**Review by FliegendeWurst — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Review by tomodachi94 — APPROVED** [review]
Lgtm, thank you ❤️

**Pull request merged**